src/OrchardCore.Modules/OrchardCore.Users/Endpoints/EmailAuthenticator/SendCode.cs

```diff
@@ -0,0 +1,116 @@
+using OrchardCore.Users.Models;
+
+namespace OrchardCore.Users.Endpoints.EmailAuthenticator;
+
+public static class SendCode
+{
+    public const string RouteName = "EmailAuthenticatorSendCode";
+
+    public static IEndpointRouteBuilder AddEmailSendCodeEndpoint<T>(this IEndpointRouteBuilder builder)
+    {
+        builder.MapPost("TwoFactor-Authenticator/EmailSendCode", HandleAsync<T>)
+            .AllowAnonymous()
+            .WithName(RouteName)
+            .DisableAntiforgery();
+
+        return builder;
+    }
+
+    private static async Task<IResult> HandleAsync<T>(
+        SignInManager<IUser> signInManager,
+        UserManager<IUser> userManager,
+        ISiteService siteService,
+        IEmailService emailService,
+        ILiquidTemplateManager liquidTemplateManager,
+        HtmlEncoder htmlEncoder,
+        IStringLocalizer<T> S)
+    {
+        var user = await signInManager.GetTwoFactorAuthenticationUserAsync();
+        var errorMessage = S["The email could not be sent. Please attempt to request the code at a later time."];
+
+        if (user == null)
+        {
+            return TypedResults.BadRequest(new
+            {
+                success = false,
+                message = errorMessage.Value,
+            });
+        }
+
+        var settings = (await siteService.GetSiteSettingsAsync()).As<EmailAuthenticatorLoginSettings>();
+        var code = await userManager.GenerateTwoFactorTokenAsync(user, TokenOptions.DefaultEmailProvider);
+
+        var to = await userManager.GetEmailAsync(user);
+        var subject = await GetSubjectAsync(settings, user, code, liquidTemplateManager, htmlEncoder);
+        var body = await GetBodyAsync(settings, user, code, liquidTemplateManager, htmlEncoder);
+        var result = await emailService.SendAsync(to, subject, body);
+
+        return TypedResults.Ok(new
+        {
+            success = result.Succeeded,
+            message = result.Succeeded ? S["A verification code has been sent via email. Please check your email for the code."].Value
+            : errorMessage.Value,
+        });
+    }
+
+    private static Task<string> GetSubjectAsync(
+        EmailAuthenticatorLoginSettings settings,
+        IUser user,
+        string code,
+        ILiquidTemplateManager liquidTemplateManager,
+        HtmlEncoder htmlEncoder)
+    {
+        var message = string.IsNullOrWhiteSpace(settings.Subject)
+        ? EmailAuthenticatorLoginSettings.DefaultSubject
+        : settings.Subject;
+
+        return GetContentAsync(message, user, code, liquidTemplateManager, htmlEncoder);
+    }
+
+    private static Task<string> GetBodyAsync(
+        EmailAuthenticatorLoginSettings settings,
+        IUser user,
+        string code,
+        ILiquidTemplateManager liquidTemplateManager,
+        HtmlEncoder htmlEncoder)
+    {
+        var message = string.IsNullOrWhiteSpace(settings.Body)
+        ? EmailAuthenticatorLoginSettings.DefaultBody
+        : settings.Body;
```

Comment:
```suggestion
            ? EmailAuthenticatorLoginSettings.DefaultBody
            : settings.Body;
```

Reply:
No because we want to follow VS placement

Reply:
Seems I need to open this or any PR in VS, to make sure